blocks/algorithms/__init__.py

```diff
@@ -657,3 +657,36 @@ def compute_step(self, param, previous_step):
         step = tensor.switch(not_finite, self.scaler * param, previous_step)
 
         return step, []
+
+
+class Restrict(StepRule):
+    """Applies a given :class:`StepRule` only to certain variables.
+
+    Example applications include clipping steps on only certain parameters,
+    or scaling a certain kind of parameter's updates (e.g. adding an
+    additional scalar multiplier to the steps taken on convolutional
+    filters).
+
+    Parameters
+    ----------
+    step_rule : :class:`StepRule`
+        The :class:`StepRule` to be applied on the given variables.
+    variables : iterable
+        A collection of Theano variables on which to apply `step_rule`.
+        Variables not appearing in this collection will not have
+        `step_rule` applied to them; their updates will be determined
```

Comment:
The last phrase of this docstring is confusing a bit since it refers to some sort of chaining which does not happen in this class. I understood that you had `CompositeRule` in mind, but I think that for a newcomer it would do more harm than good. I would remove this last phrase.